src/scene/vcModel.cpp

```diff
@@ -509,8 +509,22 @@ void vcModel::HandleContextMenu(vcState *pProgramState)
       static vcQueryNode *s_pQuery = nullptr;
 
       if (ImGui::IsWindowAppearing())
+      {
         s_pQuery = nullptr;
 
+        // Set a default file name to be exported if there isn't.
+        if (udStrlen(pProgramState->modelPath) == 0)
+        {
+          size_t fileNameLength = 0;
+          if (udStrchr(pProgramState->sceneExplorer.clickedItem.pItem->pName, ".", &fileNameLength))
+          {
+            memset(pProgramState->modelPath, 0, sizeof(pProgramState->modelPath));
+            udStrncpy(pProgramState->modelPath, pProgramState->sceneExplorer.clickedItem.pItem->pName, fileNameLength);
+            udStrncpy(pProgramState->modelPath + fileNameLength, sizeof(pProgramState->modelPath), SupportedTileTypes_ExportDefault, udStrlen(SupportedTileTypes_ExportDefault));
```

Comment:
Would recommend `udStrcat` here as its much cleaner:

```suggestion
            udStrcat(pProgramState->modelPath, SupportedTileTypes_ExportDefault);
```

Reply:
Fixed.